### PR TITLE
Adds Nanite Beds

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request
Adds Nanite Beds, a new medical machine that has its own nanite component, and applies it to patients that are buckled to it while active. All the benefits of nanites, without the committment!
They can also connect to surgery computers, like stasis beds.

The machine allows you to change the cloud ID at any time. 

The nanites inside behave much like nanites inside a mob: they have a limited, regenerating volume (although faster than base nanites) which will prevent the nanites from working if it runs out; they can be disrupted by EMPs, and they will corrupt over time if not cloud-synced.
A key difference is that these nanites are not destroyable, and at most will reach 0 volume. The bed can still be smashed the old-fashioned way.

## Why It's Good For The Game
Allows medbay to make better use of the whole range of medical nanite programs, which often go unused because of their niche utility, like blood regeneration or brain healing.

Using nanite beds is safer than getting your own nanites, of course, but they can still be tampered with to cause harm (until the target is removed from the bed), since the bed does not give a readout of the nanite programs in the selected cloud.

## Changelog
:cl:
add: Added nanite beds, researchable from techwebs.
add: Nanite beds give any patient buckled to them the effect of the nanite cloud they're linked to, without needing to inject them the normal way.
/:cl:

### CODE REVIEW

**Comment by bobbahbrown** [comment]
could a emagged version of this have the nanites stay in the person for a period of time after they depart the bed

**Review by stylemistake — APPROVED** [review]
Very cool idea, this is a viable alternative to sleepers since it requires some work from science department and makes a very good use of nanites.
Tgui good, did not test though.

**Comment by QualityVan** [comment]
If these're intended for medbay use, could medbay get a way to program them?

**Comment by ATH1909** [comment]
> could a emagged version of this have the nanites stay in the person for a period of time after they depart the bed

You shouldn't need to emag this; just set it to the cloud ID of a nanite kill setup.

**Comment by ATH1909** [comment]
Also, uh, is anyone else concerned by the fact that these are basically cryo pods, but without the unconsciousness or reagent consumption?

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 1]
```diff
@@ -749,6 +749,15 @@
...
 	needs_anchored = FALSE
 
+/obj/item/circuitboard/machine/nanobed
+	name = "Nanite Bed (Machine Board)"
+	icon_state = "medical"
+	build_path = /obj/machinery/nanobed
+	req_components = list(
+		/obj/item/stock_parts/scanning_module = 1,
+		/obj/item/stock_parts/manipulator = 1,
+		/obj/item/stock_parts/matter_bin = 1)
```
Comment (ATH1909) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm]: could you maybe add effects to these stock parts? Like, increasing the maximum nanite capacity or the nanite regen rate of the nanobed

**Comment by optimumtact** [comment]
I am very very very hesitant to add this.

I'll have to think about it, but it covers much of the same tropes that we just cut sleepers and cloning for.

Mainly that they're just timebased mechanics for healing people.

**Comment by XDTM** [comment]
In nanobeds' defense, unlike those two, they require some effort to set up, since they need to be researched and built in medbay, while someone else has to program one or more sets of useful nanite programs, and communicate them to the medbay staff that will be using the bed. 
Someone could decide to build a nanite programming setup in medbay along with the bed, but that would require even more space and effort, which at this point should be enough to offset the benefits given.

On top of this, the bed still has a limited charge, so overuse will make them grind down to a halt; especially if they're being used to treat common brute and burn damage, since bio-reconstruction is one of the more expensive programs to run.

File: code/game/objects/items/circuitboards/machine_circuitboards.dm (hunk 1 above)
Comment (XDTM) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm]: good idea, i'll add part scaling

**Comment by AdamElTablawy** [comment]
So basically someone builds a shitty heal program and you strap the guy to the bed and they just heal over time, no issue.

like, why diversify at all? toxin heal/normal nanite heal/blood regen would be sufficient. it's the normal nanite setup that most people use anyway, they diversify if they want defibs or dermal hardenin or whatever which this has no need of.

this just becomes basically a cryopod or sleeper and better than a stasis bed, and the only part of it that requires effort is about two minutes worth where a doctor says 'ai let me into nanites' and programs a 3-program nanite heal mix that treats everything

**Comment by optimumtact** [comment]
basically that

**Comment by WarlockD** [comment]
Yeah and it really dosnt add anything different than a defaut cryo setup does.  A good cryo cocktail would beat this.

This is comming from someone loves nanites and wants to expand their progamably too:p

**Comment by SkeletalElite** [comment]
Maybe make it so it requires a some kind of fluid to generate nanites
Like it has to be full of saline glucose or something and will use it up over time.

**Comment by optimumtact** [comment]
I'm going to pass on this one, sorry.

**Comment by XDTM** [comment]
Fair enough, i agree with your points after thinking about it

**Pull request closed without merging**